TodoParser.py

```diff
@@ -232,32 +247,95 @@ def parse(self, diff_file):
                                        + (r'(?!(' + '|'.join(suff_escape_list) + r'))' if len(suff_escape_list) > 0
                                           else '')
                                        + r'\s*.+$)')
-                    comments = re.finditer(comment_pattern, block['hunk'], re.MULTILINE)
```

Comment:
And here's the huge diff that I couldn't find a way to make smaller. This is the real crux of the fix to the various issues.

Previously, `self._extract_issue_if_exists()` was called both from the `marker['type'] == 'line'` block and the corresponding `else` block (for block-style comments). It's now called in just one place, after the `if/else` block. For both branches, a `contiguous_comments_and_positions` list is constructed. As before, comments which are contiguous are grouped together on the assumption that they're related to the same issue. But what's new here is that this is now _also_ tracking the position (relative to the start of the hunk) where the comment begins. Having this knowledge is what allows us to later differentiate same-titled issues within the same file.